apps/prairielearn/src/components/SyncErrorsAndWarnings.html.tsx

```diff
@@ -0,0 +1,184 @@
+      authz_data={authz_data}
+      exampleCourse={course.example_course}
+      syncErrors={question.sync_errors}
+      syncWarnings={question.sync_warnings}
+      fileEditUrl={`${urlPrefix}/question/${question.id}/file_edit/questions/${question.qid}/info.json`}
+      context={'question'}
+    />
+  );
+}
+
+export function CourseInstanceSyncErrorsAndWarnings({
+  authz_data,
+  courseInstance,
+  course,
+  urlPrefix,
+}: {
+  authz_data: { has_course_instance_permission_edit: boolean };
+  courseInstance: StaffCourseInstance;
+  course: StaffCourse;
+  urlPrefix: string;
+}) {
+  return (
+    <SyncErrorsAndWarnings
+      authz_data={authz_data}
+      exampleCourse={course.example_course}
+      syncErrors={courseInstance.sync_errors}
+      syncWarnings={courseInstance.sync_warnings}
+      fileEditUrl={`${urlPrefix}/instance_admin/file_edit/courseInstances/${courseInstance.short_name}/infoCourseInstance.json`}
+      context={'course instance'}
+    />
+  );
+}
+
+export function AssessmentSyncErrorsAndWarnings({
+  authz_data,
+  assessment,
+  courseInstance,
+  course,
+  urlPrefix,
+}: {
+  authz_data: { has_course_instance_permission_edit: boolean };
+  assessment: Assessment;
+  courseInstance: StaffCourseInstance;
+  course: StaffCourse;
+  urlPrefix: string;
+}) {
+  return (
+    <SyncErrorsAndWarnings
+      authz_data={authz_data}
+      exampleCourse={course.example_course}
+      syncErrors={assessment.sync_errors}
+      syncWarnings={assessment.sync_warnings}
+      fileEditUrl={`${urlPrefix}/assessment/${assessment.id}/file_edit/courseInstances/${courseInstance.short_name}/assessments/${assessment.tid}/infoAssessment.json`}
+      context={'assessment'}
+    />
+  );
+}
+
+function SyncErrorsAndWarnings({
+  authz_data,
+  exampleCourse,
+  syncErrors,
+  syncWarnings,
+  fileEditUrl,
+  context,
+}: {
+  authz_data: { has_course_instance_permission_edit: boolean };
+  exampleCourse: boolean;
+  syncErrors: string | null;
+  syncWarnings: string | null;
+  fileEditUrl: string;
+  context: 'course' | 'question' | 'course instance' | 'assessment';
+}) {
+  if (!authz_data.has_course_instance_permission_edit) {
+    return null;
+  }
+  if (!syncErrors && !syncWarnings) {
+    return null;
+  }
+
+  const syncErrorsAnsified = syncErrors ? ansiToHtml(syncErrors) : null;
+  const syncWarningsAnsified = syncWarnings ? ansiToHtml(syncWarnings) : null;
+  const infoFileName = fileEditUrl.split('/').pop();
+
+  return (
+    <>
+      {syncErrors ? (
+        <div class="alert alert-danger" role="alert">
+          <h2 class="h5 alert-heading">Sync error</h2>
+          <p>
+            There was an error syncing this {context}; the information you see below may be
+            inconsistent with this {context}'s <code>{infoFileName}</code> file. Please correct the
+            error and sync again.
+          </p>
+          <pre
+            class={`text-white rounded p-3 ${exampleCourse ? 'mb-0' : ''}`}
```

Comment:
Nit, consider using `clsx` here and elsewhere where you construct a `class` string.

Reply:
Changed in 2ff1900a4421ded7bd9cb14e30b56da3394e7bb2.